### PR TITLE
feat(sdk): support online serving

### PR DESCRIPTION
**Pull request by jialeicui**
## Description

SDK support online serving

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [x] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [x] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/service.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+
+class Response(ABC):
+    # TODO: support mime_type and body
+    @abstractmethod
+    def dump(self, resp: t.Any) -> bytes:
+        raise NotImplementedError()
+
+
+class GrayscaleImageRequest(Request):
```
Comment (tianweidut) [inline client/starwhale/api/_impl/service.py:23]: can we move the load function into [dataset/type.py](https://github.com/star-whale/starwhale/blob/main/client/starwhale/core/dataset/type.py) `GrayscaleImage` class?  From the user's view, the `GrayscaleImage`  can build a dataset and receive HTTP requests.

Comment (tianweidut) [reply client/starwhale/api/_impl/service.py:23]:
```
from typing_extensions import Protocol
class Request(Protocol):
    def load(self, req: t.Any) -> t.Any:
         ...
class Response(Protocol):
    def dump(self, req: t.Any) -> t.Any:
         ...
```
can we use `Protocol` to define `Request` and `Response`?

File: client/starwhale/core/model/view.py [hunk 2]
```diff
@@ -145,6 +148,19 @@ def tag(self, tags: t.List[str], remove: bool = False, quiet: bool = False) -> N
             console.print(f":surfer: add tags [red]{tags}[/] @ {self.uri}...")
             self.model.add_tags(tags, quiet)
 
+    @classmethod
+    def serve(
```
Comment (tianweidut) [inline client/starwhale/core/model/view.py:152]: add `@BaseTermView._only_standalone` for serve function?

**Review by tianweidut — COMMENTED** [review]
Others LGTM

File: client/starwhale/api/_impl/service.py (hunk 1 above)
Comment (jialeicui) [reply client/starwhale/api/_impl/service.py]: Done

File: client/starwhale/core/model/view.py (hunk 2 above)
Comment (jialeicui) [reply client/starwhale/core/model/view.py:154]: Done

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**